front/src/views/admin/Teams.vue

```diff
@@ -0,0 +1,212 @@
+            <v-icon small class="mr-2" @click="editItem(item)">
+                mdi-pencil
+            </v-icon>
+        </template>
+        <template v-slot:no-data>
+            No team 😭
+        </template>
+    </v-data-table>
+</template>
+
+<script>
+
+import firebase from 'firebase';
+
+export default {
+    name: "teams",
+    data() {
+        return {
+            teams: [],
+            organizations: [],
+            dialog: false,
+            rules: {
+                required: value => !!value || 'Required.',
+            },
+            headers: [
+                {
+                    text: 'Avatar',
+                    value: 'avatar',
+                },
+                { text: 'Name', value: 'name' },
+                { text: 'Organization', value: 'organization.name' },
+                { text: 'Actions', value: 'actions', sortable: false },
+            ],
+            editedId: null,
+            editedItem: {
+                name: '',
+                avatar: '',
+                organization: null
+            },
+            defaultItem: {
+                name: '',
+                avatar: '',
+                organization: null
+            },
+        };
+    },
+    computed: {
+        isIemValid () {
+            return this.editedItem.avatar && this.editedItem.name && this.editedItem.organization; 
+        },
+        formTitle () {
+            return this.editedId === null ? 'New Item' : 'Edit Item'
+        },
+    },
+
+    watch: {
+        dialog (val) {
+            val || this.close()
+        }
+    },
+
+    async created() {
+        await this.getOrganizations();
+        await this.getTeams();
+    },
+
+    methods: {
+
+        async getTeams() {
+            const teams = await this.concatDocument(firebase.firestore().collection('teams').get());
+            teams.forEach(team => {
+                const org = this.organizations.find(org => org.id === team.organization.id);
+                team.organization.name = org.name;
+            });
+            this.teams = teams;
+        },      
+        async getOrganizations() {
+            this.organizations = await this.concatDocument(firebase.firestore().collection('organizations').get());
+        },
+        async concatDocument(elem) {
+            return elem.then(querySnapshot => {
+                const documents = querySnapshot.docs.map(doc => {
+                    return {
+                        id: doc.id,
+                        ...doc.data(),
+                    }
+                })
+                return documents;
+            });
+        },
+
+        editItem (item) {
+            this.editedId = item.id;
+            this.editedItem = Object.assign({}, item)
+            this.dialog = true
+        },
+
+        close () {
+            this.dialog = false
+            this.$nextTick(() => {
+                this.editedItem = Object.assign({}, this.defaultItem)
+                this.editedId = null;
```

Comment:
do we need editedId field if we already have editedItem (which hold the id too)?

Reply:
I reuse the vuetify way but yes this.editedItem.id should exist